### ISSUE TITLE
new type of submodule reference

### ISSUE THREAD

**Issue by andrewseddon** [circuithub/massive-git#2]
A Git tree references a specific commit SHA1 for a sub-module. This makes the super project contain a specific revision of a sub project. 

This is quite confusing for many situations. Fro example if you've got a project which contains a bunch of components and you update a component you probably want the default behaviour to be that the component in that project is updated too.

Need to think about how to handle this. Comes under the banner of a higher level abstraction on Git submodules.

**Comment by podviaznikov** [comment]
I'm going to look into submodules in a few days.

As I understood Tree can have references to trees, blobs and commits. And references to commit is references to submodule.

**Comment by andrewseddon** [comment]
yep, which links to a specific revision of the submodule. I'm just wondering if we want to create a new type of link that always references the latest commit. It would be like saying include the head of the sub-module.

**Comment by podviaznikov** [comment]
We can do this. But than checkout of the repo by Git will be broken. Real Git will have no idea of such link.

I know that we are going to build some massive-git tool, but maybe try to be as more Git-compliant as possible.

E.x. we can save just commit as link and internally when we fetch that tree we will check for which repo this commit belongs and fetch head instead of version from the commit. So we can defined such behavior when we will read trees not save them.

However I'm not sure about this at this point. I still didn't read a lot about submodules.

**Comment by andrewseddon** [comment]
I was thinking we would export to standards compliant repo by changing to a link to the latest commit at the point at which we export.

Maybe your option of just walking up the commit chain to find the latest revision is best though.
